doc/internals/performance.md

```diff
@@ -9,18 +9,52 @@ performance even with complex Scala code.
-cases.
+The default `fastOpt` configuration is to use the Scala.js optimizer, and keep strict checks for [the undefined behaviors of Scala.js]({{ site.baseufl }}/doc/semantics.html#undefined-behaviors).
+We can see that, in development mode, Scala.js runs between around 0.9x and 6x slower than the JVM, with the majority of benchmarks in the 1.5x to 3x bracket.
+
+<div data-benchmarks="bounce brainfuck cd deltablue gcbench json kmeans list mandelbrot nbody permute richards tracer"
+  data-configs="es5.1 dev no-opt no-gcc Node.js;es5.1 dev yes-opt no-gcc Node.js"
+  data-confignames="Without optimizer;With optimizer"></div>
+
+### In production mode (fullOpt)
+
+In the `fullOpt` configuration, we remove the checks for undefined behaviors.
+Moreover, by default we enable the Google Closure Compiler (GCC) in addition to our own optimizer.
+
+In the first set of benchmarks below, we see a general trend.
+GCC tends to improve the run-time performance a bit, but not nearly as much as our own optimizer.
+The combination of both is mostly equivalent to just our own optimizer, but the benefits of GCC are mostly about code size.
+
+<div data-benchmarks="bounce brainfuck cd deltablue kmeans nbody permute richards tracer"
+  data-configs="es5.1 prod no-opt no-gcc Node.js;es5.1 prod no-opt yes-gcc Node.js;es5.1 prod yes-opt no-gcc Node.js;es5.1 prod yes-opt yes-gcc Node.js"
+  data-confignames="No optimizer, no GCC;No optimizer, with GCC;With optimizer, no GCC;With optimizer and GCC"></div>
+
+The following 4 benchmarks deviate from the general trend.
+In all cases, when both optimizers are enabled, the Scala.js benchmarks run between 0.9x and 3x slower than the JVM.
+
+<div data-benchmarks="gcbench json list mandelbrot"
+  data-configs="es5.1 prod no-opt no-gcc Node.js;es5.1 prod no-opt yes-gcc Node.js;es5.1 prod yes-opt no-gcc Node.js;es5.1 prod yes-opt yes-gcc Node.js"
+  data-confignames="No optimizer, no GCC;No optimizer, with GCC;With optimizer, no GCC;With optimizer and GCC"></div>
+
+### Comparison between fullOpt and hand-written JavaScript
+
+For three benchmarks, we also have the hand-written JavaScript code, coming from the JavaScript Octane benchmark suite.
+We compare Scala.js' production mode against the hand-written JavaScript code below.
+In the worst case, `deltablue`, fully optimized Scala.js is 1.27x slower than hand-written JavaScript.
+In the best case, `tracer`, Scala.js executes in 0.67x the time of hand-written JavaScript, i.e., 33% faster.
+
+<div data-benchmarks="deltablue richards tracer"
+  data-configs="es5.1 prod no-opt no-gcc Node.js;es5.1 prod no-opt yes-gcc Node.js;es5.1 prod yes-opt no-gcc Node.js;es5.1 prod yes-opt yes-gcc Node.js;js Node.js"
+  data-confignames="No optimizer, no GCC;No optimizer, with GCC;With optimizer, no GCC;With optimizer and GCC;Hand-written JavaScript"
+  data-y-axis-max="5.0"></div>
```

Comment:
It might be nice to present the graph normalized in the form you talk about it. That is, normalized w.r.t. normalized JavaScript (the other graph should probably also stay for comparison).